contrib/check-semver.sh

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env bash
+#
+# Checks the public API of crates, exits with non-zero if there are currently
+# changes to the public API not already committed to in the various api/*.txt
+# files.
+
+set -euo pipefail
+
+NIGHTLY=$(cat nightly-version)
+# Our docs have broken intra doc links if all features are not enabled.
+RUSTDOCFLAGS="-Z unstable-options --document-private-items --document-hidden-items --output-format=json --cap-lints=allow"
+
+if [ -n "${GITHUB_BASE_REF+x}" ]; then
+  MASTER_COMMIT=$GITHUB_BASE_REF # running on CI
+else
+  MASTER_COMMIT=$(git rev-parse master)
+fi
+
+main() {
+    need_nightly
+
+    # on current commit
+    generate_json_files_all_features "bitcoin" "current"
+    generate_json_files_no_default_features "bitcoin" "current"
+
+    generate_json_files_all_features "base58ck" "current"
+    generate_json_files_no_default_features "base58ck" "current"
+
+    generate_json_files_no_default_features "bitcoin_hashes" "current"
+    generate_json_files_features_alloc "bitcoin_hashes" "current"
+
+    generate_json_files_no_default_features "bitcoin-units" "current"
+    generate_json_files_features_alloc "bitcoin-units" "current"
+
+    generate_json_files_no_default_features "bitcoin-io" "current"
+    generate_json_files_features_alloc "bitcoin-io" "current"
+
+
+    # switch to master
+    git checkout "$MASTER_COMMIT"
+
+    generate_json_files_all_features "bitcoin" "master"
+    generate_json_files_no_default_features "bitcoin" "master"
+
+    generate_json_files_all_features "base58ck" "master"
+    generate_json_files_no_default_features "base58ck" "master"
+
+    generate_json_files_no_default_features "bitcoin_hashes" "master"
+    generate_json_files_features_alloc "bitcoin_hashes" "master"
+
+    generate_json_files_no_default_features "bitcoin-units" "master"
+    generate_json_files_features_alloc "bitcoin-units" "master"
+
+    generate_json_files_no_default_features "bitcoin-io" "master"
+    generate_json_files_features_alloc "bitcoin-io" "master"
+
+    # Check for API semver breaks
+    run_cargo_semver_check "bitcoin" "all-features"
+    run_cargo_semver_check "bitcoin" "no-default-features"
+    run_cargo_semver_check "base58ck" "all-features"
+    run_cargo_semver_check "base58ck" "no-default-features"
+    run_cargo_semver_check "bitcoin_hashes" "no-default-features"
+    run_cargo_semver_check "bitcoin_hashes" "alloc"
+    run_cargo_semver_check "bitcoin-units" "no-default-features"
+    run_cargo_semver_check "bitcoin-units" "alloc"
+    run_cargo_semver_check "bitcoin-io" "no-default-features"
+    run_cargo_semver_check "bitcoin-io" "alloc"
+
+    check_for_breaking_changes
+}
+
+# Run cargo doc
+run_cargo_doc() {
+    RUSTDOCFLAGS="$RUSTDOCFLAGS" cargo +"$NIGHTLY" doc --no-deps "$@"
```

Comment:
Please make sure to use all the flags `cargo-semver-checks` uses to generate rustdoc JSON, or else you'll get unexpected behavior from the tool like false-positives, false-negatives, and crashes. Those will be hell to debug because they won't reproduce when you run `cargo-semver-checks` directly.

https://github.com/obi1kenobi/cargo-semver-checks/blob/5108eeeffaf424e8d1a0019534414e7f11276a66/src/rustdoc_cmd.rs#L113

Reply:
Yeah check https://github.com/rust-bitcoin/rust-bitcoin/blob/264051814b858a78b96d78e45bf468b9dba7fae7/contrib/check-semver.sh#L11

Reply:
Ah I missed that, my bad!